sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java

```diff
@@ -0,0 +1,1395 @@
+        return new SubscriptExpression((Expression) visit(context.value), (Expression) visit(context.index));
+    }
+
+    @Override
+    public Node visitSubscriptSafe(SqlBaseParser.SubscriptSafeContext context) {
+        if (context.qname() != null) {
+            return new QualifiedNameReference(getQualifiedName(context.qname()));
+        }
+        return new SubscriptExpression((Expression) visit(context.value), (Expression) visit(context.index));
+    }
+
+    @Override
+    public Node visitQname(SqlBaseParser.QnameContext context) {
+        return new QualifiedNameReference(getQualifiedName(context));
+    }
+
+    @Override
+    public Node visitSimpleCase(SqlBaseParser.SimpleCaseContext context) {
+        return new SimpleCaseExpression(
+            (Expression) visit(context.valueExpression()),
+            visit(context.whenClause(), WhenClause.class),
+            visitIfPresent(context.elseExpr, Expression.class).orElse(null));
+    }
+
+    @Override
+    public Node visitSearchedCase(SqlBaseParser.SearchedCaseContext context) {
+        return new SearchedCaseExpression(
+            visit(context.whenClause(), WhenClause.class),
+            visitIfPresent(context.elseExpr, Expression.class).orElse(null));
+    }
+
+    @Override
+    public Node visitIfCase(SqlBaseParser.IfCaseContext context) {
+        return new IfExpression(
+            (Expression) visit(context.condition),
+            (Expression) visit(context.trueValue),
+            visitIfPresent(context.falseValue, Expression.class));
+    }
+
+    @Override
+    public Node visitWhenClause(SqlBaseParser.WhenClauseContext context) {
+        return new WhenClause((Expression) visit(context.condition), (Expression) visit(context.result));
+    }
+
+    @Override
+    public Node visitFunctionCall(SqlBaseParser.FunctionCallContext context) {
+        return new FunctionCall(
+            getQualifiedName(context.qname()),
+            visitIfPresent(context.over(), Window.class).orElse(null),
+            isDistinct(context.setQuant()),
+            visit(context.expr(), Expression.class));
+    }
+
+    // Literals
+
+    @Override
+    public Node visitNullLiteral(SqlBaseParser.NullLiteralContext context) {
+        return NullLiteral.INSTANCE;
+    }
+
+    @Override
+    public Node visitStringLiteral(SqlBaseParser.StringLiteralContext context) {
+        return new StringLiteral(unquote(context.STRING().getText()));
+    }
+
+    @Override
+    public Node visitIntegerLiteral(SqlBaseParser.IntegerLiteralContext context) {
+        return new LongLiteral(context.getText());
+    }
+
+    @Override
+    public Node visitDecimalLiteral(SqlBaseParser.DecimalLiteralContext context) {
+        return new DoubleLiteral(context.getText());
+    }
+
+    @Override
+    public Node visitBooleanLiteral(SqlBaseParser.BooleanLiteralContext context) {
+        return Literal.fromObject(Boolean.valueOf(context.getText()));
```

Comment:
Could return Literal.TRUE_LITERAL / FALSE_LITERAL directly instead of going through the logic in `fromObject`